setup.sh

```diff
@@ -14,215 +14,228 @@
+        touch /root/results
+        touch /root/results2
+    fi
+
+    if [[ ${#installArray[@]} -gt 0 ]]; then
+        echo_warn "Application install picker will be skipped"
+        #check Line 229 or something
+        priority=(nginx rtorrent deluge qbittorrent autodl panel vsftpd ffmpeg quota)
+        for i in "${installArray[@]}"; do
+            #shellcheck disable=SC2199,SC2076
+            if [[ " ${priority[@]} " =~ " ${i} " ]]; then
+                echo "$i" >> /root/results
+                echo_info "$i added to install queue 1"
+                touch /tmp/."$i".lock
+            else
+                echo "$i" >> /root/results2
+                echo_warn "$i added to install queue 2"
+            fi
+        done
+    fi
+}
+_option_parse "$@"
 
 _os() {
     if [ ! -d /install ]; then mkdir /install; fi
     if [ ! -d /root/logs ]; then mkdir /root/logs; fi
-    export log=/root/logs/install.log
-    # echo "Checking OS version and release ... "
     if ! which lsb_release > /dev/null; then
-        apt-get -y -qq update >> ${log} 2>&1
-        apt-get -y -qq install lsb-release >> ${log} 2>&1
+        apt_install lsb-release
     fi
     distribution=$(lsb_release -is)
-    release=$(lsb_release -rs)
     codename=$(lsb_release -cs)
     if [[ ! $distribution =~ ("Debian"|"Ubuntu") ]]; then
-        echo "Your distribution ($distribution) is not supported. Swizzin requires Ubuntu or Debian." && exit 1
+        echo_error "Your distribution ($distribution) is not supported. Swizzin requires Ubuntu or Debian."
+        exit 1
     fi
     if [[ ! $codename =~ ("xenial"|"bionic"|"stretch"|"buster"|"focal") ]]; then
-        echo "Your release ($codename) of $distribution is not supported." && exit 1
+        echo_error "Your release ($codename) of $distribution is not supported."
+        exit 1
     fi
-    # echo "I have determined you are using $distribution $release."
 }
 
 function _preparation() {
-    echo "Updating system and grabbing core dependencies."
+    echo_info "Preparing system"
+    apt_update # Do this because sometimes the system install is so fresh it's got a good stam but it is "empty"
+
     if [[ $distribution = "Ubuntu" ]]; then
-        echo "Enabling required repos"
+        echo_progress_start "Enabling required repos"
         if ! which add-apt-repository > /dev/null; then
-            apt-get install -y -q software-properties-common >> ${log} 2>&1
+            apt_install software-properties-common
         fi
         add-apt-repository universe >> ${log} 2>&1
         add-apt-repository multiverse >> ${log} 2>&1
         add-apt-repository restricted -u >> ${log} 2>&1
+        echo_progress_done
     fi
 
-    echo "Performing a system upgrade"
-    apt-get -q -y update >> ${log} 2>&1
-    apt-get -q -y upgrade >> ${log} 2>&1
+    apt_upgrade
+
+    if ! bash /etc/swizzin/scripts/update/10-dependencies.sh; then
+        echo_error "Dependencies failed to install"
```

Comment:
Should we consider exiting here if depends fail? In my experience the install is going to botch pretty hard if these don't get installed.